File: frontend/lib/app/data/models/post_model.dart
class PostModel {
  final String title;
  final String description;
  final String category;
  final bool active;
  final List<String> images;
  final int coverImage;
  final double price;

  PostModel(
      {required this.title,
      required this.description,
      required this.category,
      required this.active,
      required this.images,
      required this.coverImage,
      required this.price});

  factory PostModel.fromJson(Map<String, dynamic> json) {
    return PostModel(
      title: json['title'],
      description: json['description'],
      category: json['category'],
      active: json['active'],
      images: List<String>.from(json['images']),
      coverImage: json['coverImage'],
      price: json['price'].toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'category': category,
      'active': active,
      'images': images,
      'coverImage': coverImage,
      'price': price,
    };
  }
}

File: backend/models/PostModel.js
import { Schema, model } from "mongoose";

const postSchema = new Schema({
  title: String,
  description: String,
  category: String,
  active: Boolean,
  images: [String],
  coverImage: Number,
  price: Number,
});

postSchema.virtual("id").get(function () {
  return this._id.toString();
});

postSchema.set("toJSON", {
  virtuals: true,
  versionKey: false,
  transform: function (doc, ret) {
    delete ret._id;
  },
});

export default model("Post", postSchema);
